### PR TITLE
Move to vinta prospector fork

### LINKED ISSUE #186: Fix pylint and astroid versions
Current version specifications are conflicting in pipenv. 
```
astroid = "==1.6.1"
pylint = "==1.8.2"
```
Seems to work but weird to do such a specific pin.

### PR DESCRIPTION
**Pull request by chocoelho**
## Description

Change `prospector` dependency to use https://github.com/vintasoftware/prospector. This fork has all upstream updates with some other improvements. Resolves #186.

## Motivation and Context

While we don't decide whether to drop prospector or not, we need to be able to use the latest pylint/astroid/pycodestyle versions and we can only do that on our fork for now.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires documentation updates.
- [ ] I have updated the documentation accordingly.
- [x] My change requires dependencies updates.
- [x] I have updated the dependencies accordingly.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -33,9 +33,9 @@ opbeat = "*"
 whitenoise = "*"
 bandit = "*"
 coverage = "*"
-astroid = "<1.6"
-pylint = "<1.8"
-prospector = {extras = ["with-vulture"]}
+astroid = "*"
+pylint = "*"
```
Comment (hugobessa) [inline Pipfile]: Wouldn't be better to remove them from our Pipfile, since they're prospector deps (astroid is a pylint dep, to be more precise)?

Comment (chocoelho) [reply Pipfile]: Correct. This way we don't need to deal with these dependencies here.

**Comment by chocoelho** [comment]
This is a guarantee that we have something working while `prospector` does not move to PyCQA.

**Comment by chocoelho** [comment]
Resolves #186

**Pull request merged**